### PR TITLE
Auto-print returned values from Lua macros

### PR DESCRIPTION
**Pull request by pmatilai**
Avoids having to manually "print()" everything, at the cost of requiring
a "return" which is not really less to type but more natural for
programming, and other benefits likely to follow.

### CODE REVIEW

**Comment by mlschroe** [comment]
Somewhat unrelated, but why doesn't `rpm_print` use `luaL_tolstring`? That also seems to be what the internal "tostring" function does.

**Comment by pmatilai** [comment]
Yup, noticed the same thing: https://github.com/rpm-software-management/rpm/commit/d41143cb5f6d88eb6e8bd999ad5ea2992bfb10f7

**Comment by mlschroe** [comment]
And the `(void) fputs(s, stdout);` should be `(void) fwrite(s, 1, sl, stdout)` so that it can print binary data like the lua print() command...

File: rpmio/rpmlua.c [hunk 1]
```diff
@@ -324,12 +325,17 @@ int rpmluaRunScript(rpmlua _lua, const char *script, const char *name,
...
+    if (lua_pcall(L, 2, LUA_MULTRET, 0) != 0) {
 	rpmlog(RPMLOG_ERR, _("lua script failed: %s\n"),
 		 lua_tostring(L, -1));
 	lua_pop(L, 1);
 	goto exit;
     }
+
+    int nret = lua_gettop(L) - otop;
+    if (nret > 0 && lua->printbuf)
+	rpm_print(L);
```
[the remark this answers is not in the document]
Comment (pmatilai) [reply rpmio/rpmlua.c:337]: Right, I was wondering about the "common practise" of doing such a thing and that maybe the beef of rpm_print() should be refactored out to better faciliate calling from C or something.

[the remark this answers is not in the document]
Comment (mlschroe) [reply rpmio/rpmlua.c:337]: My suggestion was something like (untested):
```
lua_getglobal(L, "print");
lua_insert(L, otop);
if (lua_pcall(L, nret, 0, 0) != 0) {
        rpmlog(RPMLOG_ERR, result print failed: %s\n"),
                 lua_tostring(L, -1));
        lua_pop(L, 1);
}
```

File: rpmio/rpmlua.c [hunk 2]
```diff
@@ -327,12 +328,24 @@ int rpmluaRunScript(rpmlua _lua, const char *script, const char *name,
...
     }
+
+    int nret = lua_gettop(L) - otop;
+    if (nret > 0 && lua->printbuf) {
+	lua_getglobal(L, "print");
+	lua_insert(L, -(nret + 1));
+	if (lua_pcall(L, nret, 0, 0) != 0) {
+	    rpmlog(RPMLOG_ERR, _("result print failed: %s\n"),
+		    lua_tostring(L, -1));
+	    lua_pop(L, 1);
```
Comment (mlschroe) [inline rpmio/rpmlua.c:345]: `goto exit;` ?

**Pull request merged**